File: ColorfulBubbles/Color.java
import processing.core.PApplet;

public class Color {

  public static PApplet p;

  public static void init(PApplet context) {
    p = context;
  }

  public static int getRandomColor(String name) {
    int c;
    int r;
    int g;
    int b;

    switch (name) {
    case "magenta":
      r = (int)p.random(100, 255);
      g = (int)p.random(50, 100);
      b = (int)p.random(100, 255);
      break;
    case "cyan":
      r = (int)p.random(50, 100);
      g = (int)p.random(100, 255);
      b = (int)p.random(100, 255);
      break;
    case "blue":
      r = (int)p.random(50, 100);
      g = (int)p.random(50, 100);
      b = (int)p.random(100, 255);
      break;
    case "red":
      r = (int)p.random(100, 255);
      g = (int)p.random(50, 100);
      b = (int)p.random(50, 100);
      break;
    case "green":
      r = (int)p.random(50, 100);
      g = (int)p.random(100, 255);
      b = (int)p.random(50, 100);
      break;
    default:
      r = (int)p.random(50, 100);
      g = (int)p.random(50, 100);
      b = (int)p.random(50, 100);
      break;
    }

    c = p.color(r, g, b);
    return c;
  }
}

File: docs/ColorfulBubbles/color.js
// Colorful Bubbles
// Color functions

function getColorNames() {
    return ["magenta", "cyan", "blue", "red", "green"];
}

function getRandomColor(name) {
    var c;
    var r;
    var g;
    var b;

    switch (name) {
        case "magenta":
            r = random(100, 255); g = random(50, 100); b = random(100, 255);
            break;

        case "cyan":
            r = random(50, 100); g = random(100, 255); b = random(100, 255);
            break;

        case "blue":
            r = random(50, 100); g = random(50, 100); b = random(100, 255);
            break;

        case "red":
            r = random(100, 255); g = random(50, 100); b = random(50, 100);
            break;

        case "green":
            r = random(50, 100); g = random(100, 255); b = random(50, 100);
            break;

        default:
            r = random(50, 100); g = random(50, 100); b = random(50, 100);
            break;
    }

    c = color(r, g, b);
    return c;
}
